ports/atmel-samd/mphalport.c

```diff
@@ -75,6 +103,24 @@ void mp_hal_delay_us(mp_uint_t delay) {
     tick_delay(delay);
 }
 
+// Do a simple timing loop to wait for a certain number of microseconds.
+// Can be used when interrupts are disabled, which makes tick_delay() unreliable.
+//
+// Testing done at 48 MHz on SAMD21 and 120 MHz on SAMD51, multiplication and division cancel out.
+// But get the frequency just in case.
+#ifdef SAMD21
+#define DELAY_LOOP_ITERATIONS_PER_US ( (10U*48000000U) / common_hal_mcu_processor_get_frequency())
+#endif
+#ifdef SAMD51
+#define DELAY_LOOP_ITERATIONS_PER_US ( (30U*120000000U) / common_hal_mcu_processor_get_frequency())
+#endif
+
+void mp_hal_delay_us_loop(uint32_t us) {
```

Comment:
No need to add another API. Just make `mp_hal_delay_us` this implementation. It shouldn't be used for >1ms timing.